### PR TITLE
Fix vulnerable components and outdated components

### PR DESCRIPTION
**Pull request by vanderaj**
I decided since we are accepting redos fixes, here's a PR to take all known updates to eliminate known vulnerabilities and outdated components. Upon testing, there was a warning, so this PR also fixes that by providing what would have been the defaults in earlier versions of express-session to not change behavior.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -8,41 +8,41 @@
...
         "cross-env": "^7.0.2",
-        "eslint": "^7.25.0",
+        "eslint": "^7.32.0",
         "eslint-config-prettier": "^8.3.0",
-        "eslint-plugin-prettier": "^3.4.0",
-        "nodemon": "^2.0.4",
-        "pm2": "^4.5.6",
-        "prettier": "^2.2.1"
+        "eslint-plugin-prettier": "^3.4.1",
+        "mongoose": "^5.13.7",
```
Comment (SayakMukhopadhyay) [inline backend/package.json:41]: Why did you move mongoose and the aggregate-paginate libs to dev dependency?

File: backend/server.js [hunk 2]
```diff
@@ -70,7 +70,9 @@ app.use(
     name: 'EliteBGS',
     secret: secrets.session_secret,
     cookie: { maxAge: 7 * 24 * 60 * 60 * 1000 },
-    store: new mongoStore({ mongooseConnection: mongoose.connection })
+    store: new mongoStore({ mongooseConnection: mongoose.connection }),
+    resave: true,
```
Comment (SayakMukhopadhyay) [inline backend/server.js:74]: Not sure what they do but seems fine

**Review by SayakMukhopadhyay — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 1 above)
Comment (vanderaj) [reply backend/package.json:39]: it was because I was using --savedev and that puts them there. Moved back in 
ced7639
